sql/2024/04/image-formats.sql

```diff
@@ -0,0 +1,92 @@
+DECLARE
+  DATE_TO_QUERY DATE DEFAULT '2024-03-01';
+
+CREATE TEMPORARY FUNCTION
+  IS_CMS(technologies ARRAY<STRUCT<technology STRING,
+                                   categories ARRAY<STRING>,
+                                   info ARRAY<STRING>>>,
+         cms STRING,
+         version STRING)
+  RETURNS BOOL AS ( EXISTS(
+  SELECT
+    *
+  FROM
+    UNNEST(technologies) AS technology,
+    UNNEST(technology.info) AS info
+  WHERE
+      technology.technology = cms
+    AND ( version = ""
+    OR ENDS_WITH(version, ".x")
+            AND (STARTS_WITH(info, RTRIM(version, "x"))
+        OR info = RTRIM(version, ".x"))
+    OR info = version ) ) );
+
+CREATE TEMPORARY FUNCTION
+  IS_IMAGE (summary STRING)
+  RETURNS BOOLEAN AS (STARTS_WITH(LOWER(CAST(JSON_EXTRACT_SCALAR(summary, "$.mimeType") AS STRING)), 'image/'));
+
+WITH
+  pagesWithLcpImages AS (
+    SELECT
+      date,
+      client,
+      page,
+      JSON_EXTRACT_SCALAR(custom_metrics, '$.performance.lcp_elem_stats.url') AS url,
+      JSON_EXTRACT_SCALAR(custom_metrics, '$.performance.lcp_elem_stats.naturalWidth') AS image_width,
+      JSON_EXTRACT_SCALAR(custom_metrics, '$.performance.lcp_elem_stats.naturalHeight') AS image_height,
+    FROM
+      `httparchive.all.pages`
+    WHERE
+      IS_CMS(technologies,
+             'WordPress',
+             '')
+      AND LOWER(JSON_EXTRACT_SCALAR(custom_metrics, '$.performance.lcp_elem_stats.nodeName')) = 'img'
```

Comment:
Would it make sense to also include `picture` here? Granted this is likely to not be very common.

Reply:
`<picture>` still has an underlying `<img>` which is what's used. Can you have a `<picture>` without an `<img>`?

Reply:
Good point. It does require `img`: https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element

Reply:
Sounds reasonable, I'll look into it.